routing/payment_lifecycle.go

```diff
@@ -279,8 +281,28 @@ func (p *paymentLifecycle) createNewPaymentAttempt() (lnwire.ShortChannelID,
 	// with the htlcAdd message that we send directly to the
 	// switch.
 	onionBlob, c, err := generateSphinxPacket(
-		route, p.payment.PaymentHash[:], sessionKey,
+		rt, p.payment.PaymentHash[:], sessionKey,
 	)
+
+	// With SendToRoute, it can happen that the route exceeds protocol
+	// constraints. Mark the payment as failed with an internal error.
+	if err == route.ErrMaxRouteHopsExceeded ||
+		err == sphinx.ErrMaxRoutingInfoSizeExceeded {
+
+		log.Debugf("Invalid route provided for "+
+			"payment %x: %v", p.payment.PaymentHash,
+			err)
+
+		controlErr := p.router.cfg.Control.Fail(
+			p.payment.PaymentHash,
+			channeldb.FailureReasonError,
+		)
```

Comment:
nit: those two statements are both wrapped a bit too early and could be more compact if some line breaks are removed.